### PR TITLE
typed data support

### PR DESCRIPTION
**Pull request by Ptroger**
### Support Typed Data in Auth Server

Auth server was not accepting Typed Data as valid Request. Also, AuthServer stores all requests in a Postgresql DB. So this PR does three things:
- Adds TypedData as a valid input
- Changed EIP712TypedData schema so that it automatically parses message field back to an object if its a string
- Added a SerializedEIP712TypedData schema that will serialize message into a string in AuthServer upon receiving an Authorization Request.

### CODE REVIEW

File: apps/armory/src/orchestration/__test__/e2e/authorization-request.spec.ts [hunk 1]
```diff
@@ -101,6 +101,65 @@ describe('Authorization Request', () => {
...
+        typedData
+      }
+
+      const payload = {
+        authentication,
+        approvals,
+        request: req
+      }
+
+      Eip712TypedData.parse(payload.request.typedData)
```
Comment (wcalderipe) [inline apps/armory/src/orchestration/__test__/e2e/authorization-request.spec.ts:144]:
```suggestion
```

File: apps/armory/src/orchestration/persistence/schema/request.schema.ts [hunk 2]
```diff
@@ -10,17 +10,20 @@ import {
...
 ])
 
 export const createRequestSchema = z.discriminatedUnion('action', [
   createSignTransactionSchema,
   createSignMessageSchema,
   createGrantPermissionSchema,
-  createSignRawSchema
+  createSignRawSchema,
+  createSignTypedDataSchema
 ])
```
Comment (wcalderipe) [inline apps/armory/src/orchestration/persistence/schema/request.schema.ts:29]: Have you ever looked at this file before https://github.com/narval-xyz/armory/blob/feb8eba0a8470eb8d6852249c7dd27a48d7ce96b/apps/armory/src/orchestration/persistence/decode/__test__/unit/authorization-request.decode.spec.ts ?

Comment (Ptroger) [reply apps/armory/src/orchestration/persistence/schema/request.schema.ts:29]: Yes, I could add one there to check the serialization of typed data. Thought this was a bit unnecessary because this ground is already covered by the e2e test, and I assumed these were written when e2e tests where not possible. Are you pointing me to this file so that I add coverage for typed data ?

Comment (wcalderipe) [reply apps/armory/src/orchestration/persistence/schema/request.schema.ts:32]: Is the E2E covering a failing scenario or many failing scenarios? As far as I remember, the authorization request E2E tests are only testing the happy path. 

We shouldn't write tests for every single edge case as E2E because they're expensive to write, run, and maintain. This is where unit and integration tests can be very valuable, allowing us to extensively test many possible scenarios.

In the E2E always add the happy and maybe a sad path for whatever you're working on. If you have to cover many failure scenarios, it's advisable to test them as units.
